### PR TITLE
DDP-5312 part 1: support kit reasons

### PR DESCRIPTION
**Pull request by yufengwng**
## Context

Part of DDP-5312, more work will follow in subsequent PRs. This PR adds initial support for "kit reason types". These reasons are provided by DSM when it triggers a kit event. Studies decide what these reason types mean by leveraging them in their configurations. For example, TestBoston may choose to ignore events for `REPLACEMENT` kits.

## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [x] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [ ] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [x] They are user-visible in dev as a regular user journey and require no additional instructions.
- [ ] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [x] I have written automated positive tests
- [ ] I have written automated negative tests
- [ ] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [x] These changes require no special release procedures--just code!
- [ ] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

### CODE REVIEW

File: pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/KitReasonType.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package org.broadinstitute.ddp.model.dsm;
+
+public enum KitReasonType {
```
Comment (yufengwng) [inline pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/KitReasonType.java:6]: The kit reasons are still not really final. Also waiting for input from @pegahtah.

Comment (yufengwng) [reply pepper-apis/src/main/java/org/broadinstitute/ddp/model/dsm/KitReasonType.java:6]: Update: kit reasons mostly finalized.

**Comment by yufengwng** [comment]
Breaking change because we require a change in DSM side to pass us `kitReasonType`.

**Comment by yufengwng** [comment]
CC @pegahtah @SimoneMaiwald

**Comment by yufengwng** [comment]
This is ready for review. @MocanaAtBroad @ssettipalli

**Review by MocanaAtBroad — APPROVED** [review]

**Pull request merged**